File: M/1266-minimum-time-visiting-all-points/main.go
package main

import "fmt"

func minTimeToVisitAllPoints(points [][]int) int {
	if len(points) <= 1 {
		return 0
	}
	sum := 0
	x1, y1 := points[0][0], points[0][1]
	for i := 1; i < len(points); i++ {
		x2, y2 := points[i][0], points[i][1]
		divx, divy := x1-x2, y1-y2
		if divx < 0 {
			divx = -divx
		}
		if divy < 0 {
			divy = -divy
		}
		v := divx
		if divx < divy {
			v = divy
		}
		sum += v
		x1, y1 = x2, y2
	}
	return sum
}

func main() {
	fmt.Println("a")
}

File: M/1266-minimum-time-visiting-all-points/main.rs
impl Solution {
  pub fn min_time_to_visit_all_points(points: Vec<Vec<i32>>) -> i32 {
    points
      .iter()
      .fold((0, &points[0]), |(sum, prev), point| {
        (
          sum + (point[0] - prev[0]).abs().max((point[1] - prev[1]).abs()),
          point,
        )
      })
      .0
  }
}
